src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationService.kt

```diff
@@ -39,10 +41,23 @@ class VisitNotificationService(
     }
   }
 
+  private fun isNotADeleteEvent(nonAssociationChangedNotification: NonAssociationChangedNotificationDto): Boolean {
+    val nonAssociations = prisonerService.getOffenderNonAssociationList(nonAssociationChangedNotification.prisonerNumber)
+    return nonAssociations.any { it.offenderNonAssociation.offenderNo == nonAssociationChangedNotification.nonAssociationPrisonerNumber }
+  }
```

Comment:
Should I also check the dates, just dont know if they will exisit on a delete event,.. only one way to find out here,... also I dont know if there will be a race condition between this event updating the prison service and doing the update,... here... only one way to find out

Reply:
Visit notification service should be listening for the booking create/update/delete events which have been generated by the visit scheduler - then sending out notifications accordingly.

Visit notifications should be an independent service - not part of the visit scheduler!

Reply:
good point. I think the getOffenderNonAssociationList will return the current non associations while the event could be for a future date. We will have to check the API docs.

Reply:
added extra checks